staging/src/k8s.io/kubectl/pkg/cmd/create/create.go

```diff
@@ -46,6 +46,124 @@ import (
 	"k8s.io/kubectl/pkg/util/templates"
 )
 
+// CreateFlags directly reflect the information that CLI is gathering via flags. They will be converted to Options,
+// which reflect the runtime requirements for the command.
+type CreateFlags struct {
+	Factory     cmdutil.Factory
+	PrintFlags  *genericclioptions.PrintFlags
+	RecordFlags *genericclioptions.RecordFlags
+	Recorder    genericclioptions.Recorder
+
+	fieldManager string
+
+	FilenameOptions  resource.FilenameOptions
+	Selector         string
+	EditBeforeCreate bool
+	Raw              string
+
+	genericclioptions.IOStreams
+}
+
+// NewCreateFlags returns a default CreateFlags
+func NewCreateFlags(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *CreateFlags {
+	return &CreateFlags{
+		Factory:     f,
+		PrintFlags:  genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme),
+		RecordFlags: genericclioptions.NewRecordFlags(),
+
+		Recorder: genericclioptions.NoopRecorder{},
+
+		IOStreams: ioStreams,
+	}
+}
+
+// AddFlags registers flags for a cli
+func (flags *CreateFlags) AddFlags(cmd *cobra.Command) {
+	flags.PrintFlags.AddFlags(cmd)
+	flags.RecordFlags.AddFlags(cmd)
+
+	cmdutil.AddFilenameOptionFlags(cmd, &flags.FilenameOptions, "to use to create the resource")
+	cmdutil.AddValidateFlags(cmd)
+	cmdutil.AddApplyAnnotationFlags(cmd)
+	cmdutil.AddDryRunFlag(cmd)
+	cmdutil.AddFieldManagerFlagVar(cmd, &flags.fieldManager, "kubectl-create")
+
+	cmd.Flags().BoolVar(&flags.EditBeforeCreate, "edit", flags.EditBeforeCreate, "Edit the API resource before creating")
```

Comment:
Let's get these move to util functions like above please.

Reply:
Hmm, I am not sure which one, is that `cmd.Flags().BoolVar()`? so I would have to create functions in `util` like `AddBoolVarFlags(cmd, "msg")`?